### PR TITLE
feat: bench merkle proofs verification gas usage

### LINKED ISSUE #174: Gas benches for Merkle Proof Verification
Add benches based on the work in #168

### PR DESCRIPTION
**Pull request by alexfertel**
Resolves #174 

Screenshot 2024-07-15 at 16 45 37

### CODE REVIEW

File: benches/src/merkle_proofs.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    .map(|h| h.into())
+    .to_vec();
+
+    let receipts = vec![(
+        "verify()",
+        receipt!(contract.verify(proof, root.into(), leaf.into()))?,
+    )];
+
+    // Calculate the width of the longest function name.
+    let max_name_width = receipts
```
Comment (bidzyyys) [inline benches/src/merkle_proofs.rs:80]: Nit: each benchmark has the same way for calculating `max_name_width`. How about common code?

Comment (bidzyyys) [reply benches/src/merkle_proofs.rs:80]: And printing the table if applicable.

**Review by bidzyyys — APPROVED** [review]
I like it 🚀

File: benches/src/merkle_proofs.rs (hunk 1 above)
Comment (qalisander) [reply benches/src/merkle_proofs.rs:80]: Agree. Would be more simple to add another benchmarks. At least considering performance while designing new contract

**Review by qalisander — APPROVED** [review]
Good job!

File: benches/src/merkle_proofs.rs (hunk 1 above)
Comment (alexfertel) [reply benches/src/merkle_proofs.rs:80]: Yes! I'm leaving that for a separate PR, cause benchmarks are a bigger priority. I opened an issue for it some time ago https://github.com/OpenZeppelin/rust-contracts-stylus/issues/170

**Pull request merged**